functions/v2/helloworld/hello_pubsub_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_pubsub_unit_test_v2]
+
+package helloworld
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+)
+
+func TestHelloPubSub(t *testing.T) {
+	tests := []struct {
+		data string
+		want string
+	}{
+		{want: "Hello, World!\n"},
+		{data: "Go", want: "Hello, Go!\n"},
+	}
+	for _, test := range tests {
+		r, w, _ := os.Pipe()
+		log.SetOutput(w)
+		originalFlags := log.Flags()
+		log.SetFlags(log.Flags() &^ (log.Ldate | log.Ltime))
+
+		m := PubSubMessage{
+			Data: []byte(test.data),
+		}
+		cem := CloudEventMessage{
```

Comment:
issue: In the CloudEvent object, the "body" field is called `data`. Within that, the Pub/Sub Message might have it's own nested layers of data or method properties. I expect once the function is switched to CloudEvent, this test will fail.

If it does not fail for some reason, I suggest the test is brittle because it doesn't align with the expected data structures.

Reply:
The example in the documentation has `message` as the top level field.  https://cloud.google.com/eventarc/docs/cloudevents#pubsub

Reply:
I tested this manually by deploying and this data structure is correct.  It does work.

Reply:
![image](https://user-images.githubusercontent.com/55413432/128559778-2c14d87c-aaa8-48e8-a927-6700d039ce5d.png)

Reply:
I can't find where the body field is called data either..